src/game/crossteaser/mod.rs

```diff
@@ -53,31 +53,586 @@ const ABOUT: &str = "PLACEHOLDER";
 
 /* GAME IMPLEMENTATION */
 
-/// Encodes the state of a piece in the game board. For reference, a cube has
-/// six faces (up, down, etc.), and a cube with face A on top can be oriented
-/// in one of four ways (north, south, etc.).
-enum Face {
-    Up(Orientation),
-    Down(Orientation),
-    Left(Orientation),
-    Right(Orientation),
-    Front(Orientation),
-    Back(Orientation),
-    None,
+/// Encodes the state of a single piece on the game board. It achieves this by
+/// storing a value from 0-5 for each of 3 faces. This allows us to determine
+/// the exact orientation out of 24 possible. We can think of each number from
+/// 0-5 as being one of the 6 possible colors for cross faces. The faces are
+/// arranged such that each pair of opposite faces always sum to 5. Example:
+///       2
+///       |
+///   1 - 0 - 4     (5 on back)
+///       |
+///       3
+/// The above orientation will be defined as the "default" state
+/// And is at index 0 in the ORIENTATION_MAP array
+/// All 24 orientations will be some rotation of this structure. The relative
+/// positions of faces does not change.
+struct Orientation {
+    front: u64,
+    top: u64,
+    right: u64,
 }
 
-/// Encodes the orientation information about each piece in the game. Since each
-/// piece is cube-like, it is not enough to just have a face, since a cube with
-/// its "Front" face up could still be oriented in one of four ways.
-enum Orientation {
-    North,
-    East,
-    South,
-    West,
+/// Unhashed representation of a game state
+/// It is simply a vector of (# pieces) Orientations
+/// and an integer representing the location of the free space
+struct UnhashedState {
+    pieces: Vec<Orientation>,
```

Comment:
Work out the maximum length of `pieces` as allowed by a 64-bit `State`, and use a statically sized array here to allow for a no-malloc implementation.

Reply:
Since we've moved to a bitvec State implementation, should I still do this? I suppose I could impose a state size limit, given that we realistically can't solve anything bigger than 3x3 atm.